app/src/main/java/com/example/househomey/scanner/SNImageScanner.java

```diff
@@ -71,16 +75,34 @@ public void scanImage(String imageUri) {
      * @return the chosen serial number
      */
     private static String selectBestLine(Text text) {
+
         String bestElement = "";
+        List<String> strings = new ArrayList<>();
+        Pattern pattern = Pattern.compile("\\d+");
+        int maxConsecutiveCount = 0;
 
+        //Get all the text elements/words from the image as strings
         for (Text.TextBlock block : text.getTextBlocks()) {
             for (Text.Line line : block.getLines()) {
                 for (Text.Element element : line.getElements())
-                    // Serial numbers are usually at the bottom
-                    if (element.getText().length() >= bestElement.length())
-                        bestElement = element.getText();
+                    strings.add(element.getText());
+            }
+        }
+
+        //Select the string that has the most consecutive numbers on it
```

Comment:
I think maybe we should select the string with just the most numbers in it instead?